### PR TITLE
[v11] Update Firestore SPM binary to pre_rc1

### PR DESCRIPTION
**Pull request by ncooke3**
I'm expecting this to fix outstanding SPM CI issues related to Firestore. 

#no-changelog

### CODE REVIEW

**Comment by ncooke3** [comment]
Hmm, something else is broken. Seems to be related to Analytics binaries. I double checked that they are all static archives.

```
ld: warning: Could not find or use auto-linked framework 'GoogleAppMeasurement': framework 'GoogleAppMeasurement' not found
ld: Undefined symbols:
  _APMAnalyticsConfiguration, referenced from:
      +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMAppMeasurementOriginFirebase, referenced from:
      +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMConsentSettings3P, referenced from:
      +[FIRAnalytics setConsent:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMFormattedEventName, referenced from:
      +[FIRAnalytics logEventWithOrigin:name:parameters:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMFormattedUserPropertyName, referenced from:
      +[FIRAnalytics setUserPropertyString:forName:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMIsAnalyticsCollectionDeactivated, referenced from:
      +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMIsAnalyticsCollectionEnabled, referenced from:
      +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMMonitorLogTagOptionKey, referenced from:
      +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMUserDataFieldEmailAddress, referenced from:
      +[FIRAnalytics initiateOnDeviceConversionMeasurementWithEmailAddress:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMUserDataFieldHashedEmailAddress, referenced from:
      +[FIRAnalytics initiateOnDeviceConversionMeasurementWithHashedEmailAddress:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMUserDataFieldHashedPhoneNumber, referenced from:
      +[FIRAnalytics initiateOnDeviceConversionMeasurementWithHashedPhoneNumber:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _APMUserDataFieldPhoneNumber, referenced from:
      +[FIRAnalytics initiateOnDeviceConversionMeasurementWithPhoneNumber:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _FIRFirebaseVersion, referenced from:
      +[FIRAnalytics topLevelVersion] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _FIRInstallationIDDidChangeNotification, referenced from:
      +[FIRAnalytics observeFirebaseInstallationIDChanges] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _OBJC_CLASS_$_APMAnalytics, referenced from:
       in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _OBJC_CLASS_$_APMConditionalUserProperty, referenced from:
      _OBJC_CLASS_$_FIRAConditionalUserProperty in FirebaseAnalytics[arm64][4](FIRAConditionalUserProperty.o)
  _OBJC_CLASS_$_APMConditionalUserPropertyController, referenced from:
      _OBJC_CLASS_$_FIRAConditionalUserPropertyController in FirebaseAnalytics[arm64][5](FIRAConditionalUserPropertyController.o)
  _OBJC_CLASS_$_APMEvent, referenced from:
      _OBJC_CLASS_$_FIRAEvent in FirebaseAnalytics[arm64][7](FIRAEvent.o)
  _OBJC_CLASS_$_APMIdentity, referenced from:
       in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _OBJC_CLASS_$_APMMeasurement, referenced from:
       in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _OBJC_CLASS_$_APMValue, referenced from:
      _OBJC_CLASS_$_FIRAValue in FirebaseAnalytics[arm64][14](FIRAValue.o)
  _OBJC_CLASS_$_FIRApp, referenced from:
       in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _OBJC_CLASS_$_FIRComponent, referenced from:
       in FirebaseAnalytics[arm64][11](FIRAnalyticsConnector.o)
  _OBJC_CLASS_$_FIRInstallations, referenced from:
       in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _OBJC_METACLASS_$_APMConditionalUserProperty, referenced from:
      _OBJC_METACLASS_$_FIRAConditionalUserProperty in FirebaseAnalytics[arm64][4](FIRAConditionalUserProperty.o)
  _OBJC_METACLASS_$_APMConditionalUserPropertyController, referenced from:
      _OBJC_METACLASS_$_FIRAConditionalUserPropertyController in FirebaseAnalytics[arm64][5](FIRAConditionalUserPropertyController.o)
  _OBJC_METACLASS_$_APMEvent, referenced from:
      _OBJC_METACLASS_$_FIRAEvent in FirebaseAnalytics[arm64][7](FIRAEvent.o)
  _OBJC_METACLASS_$_APMValue, referenced from:
      _OBJC_METACLASS_$_FIRAValue in FirebaseAnalytics[arm64][14](FIRAValue.o)
  _kFIRLoggerAnalytics, referenced from:
      +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Review by paulb777 — APPROVED** [review]

**Comment by ncooke3** [comment]
```
ld: Undefined symbols:
  _FIRFirebaseVersion, referenced from:
      +[FIRAnalytics topLevelVersion] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _FIRInstallationIDDidChangeNotification, referenced from:
      +[FIRAnalytics observeFirebaseInstallationIDChanges] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _GULIsLoggableLevel, referenced from:
      -[APMMonitor isLoggableLevel:] in GoogleAppMeasurement[arm64][69](APMMonitor.o)
  _GULOSLogBasic, referenced from:
      -[APMASLLogger logMessage:logTag:messageCode:withLogLevel:] in GoogleAppMeasurement[arm64][12](APMASLLogger.o)
  _GULOSLogError, referenced from:
      -[APMPersistentDictionary initWithFileName:] in GoogleAppMeasurement[arm64][262](APMPersistentDictionary.o)
      _APMWriteDictionaryToURL in GoogleAppMeasurement[arm64][262](APMPersistentDictionary.o)
      -[APMUserDefaults synchronize] in GoogleAppMeasurement[arm64][316](APMUserDefaults.o)
  _GULOSLogInfo, referenced from:
      ___44+[UIViewController(APMScreenClassName) load]_block_invoke in GoogleAppMeasurement[arm64][335](UIViewController+APMScreenClassName.o)
  _GULOSLogWarning, referenced from:
      +[APMMeasurement sharedInstance] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMPersistentDictionary objectForKey:] in GoogleAppMeasurement[arm64][262](APMPersistentDictionary.o)
      -[APMPersistentDictionary setObject:forKey:] in GoogleAppMeasurement[arm64][262](APMPersistentDictionary.o)
      -[APMUserDefaults objectForKey:] in GoogleAppMeasurement[arm64][316](APMUserDefaults.o)
      -[APMUserDefaults setObject:forKey:] in GoogleAppMeasurement[arm64][316](APMUserDefaults.o)
      -[APMUserDefaults synchronize] in GoogleAppMeasurement[arm64][316](APMUserDefaults.o)
      ___44+[UIViewController(APMScreenClassName) load]_block_invoke in GoogleAppMeasurement[arm64][335](UIViewController+APMScreenClassName.o)
      ...
  _GULSetLoggerLevel, referenced from:
      -[APMMonitor setDebugModeEnabled:] in GoogleAppMeasurement[arm64][69](APMMonitor.o)
      -[APMMonitor setDebugModeEnabled:] in GoogleAppMeasurement[arm64][69](APMMonitor.o)
      -[APMMonitor setVerboseLoggingEnabled:] in GoogleAppMeasurement[arm64][69](APMMonitor.o)
  _OBJC_CLASS_$_FIRApp, referenced from:
       in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _OBJC_CLASS_$_FIRComponent, referenced from:
       in FirebaseAnalytics[arm64][11](FIRAnalyticsConnector.o)
  _OBJC_CLASS_$_FIRInstallations, referenced from:
       in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _OBJC_CLASS_$_GULAppDelegateSwizzler, referenced from:
       in GoogleAppMeasurement[arm64][7](APMAnalytics.o)
  _OBJC_CLASS_$_GULAppEnvironmentUtil, referenced from:
       in GoogleAppMeasurement[arm64][5](APMAEU.o)
  _OBJC_CLASS_$_GULMutableDictionary, referenced from:
       in GoogleAppMeasurement[arm64][35](APMEExperiment.o)
  _OBJC_CLASS_$_GULNetwork, referenced from:
       in GoogleAppMeasurement[arm64][7](APMAnalytics.o)
  _OBJC_CLASS_$_GULSwizzler, referenced from:
       in GoogleAppMeasurement[arm64][335](UIViewController+APMScreenClassName.o)
  _kFIRLoggerAnalytics, referenced from:
      +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)
  _kGULNetworkHTTPStatusCodeCannotAcceptTraffic, referenced from:
      -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
      -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
  _kGULNetworkHTTPStatusCodeNotFound, referenced from:
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
  _kGULNetworkHTTPStatusCodeNotModified, referenced from:
      ___37-[APMMeasurement retrievePSMResponse]_block_invoke in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement(SKAN) handleSKANFetchResponse:data:error:] in GoogleAppMeasurement[arm64][67](APMMeasurement+SKAN.o)
      ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement[arm64][68](APMMeasurement+URL.o)
  _kGULNetworkHTTPStatusCodeUnavailable, referenced from:
      -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
      -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
  _kGULNetworkHTTPStatusNoContent, referenced from:
      -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
      -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      ___37-[APMMeasurement retrievePSMResponse]_block_invoke in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement[arm64][68](APMMeasurement+URL.o)
  _kGULNetworkHTTPStatusOK, referenced from:
      -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
      -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      ___37-[APMMeasurement retrievePSMResponse]_block_invoke in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement(SKAN) handleSKANFetchResponse:data:error:] in GoogleAppMeasurement[arm64][67](APMMeasurement+SKAN.o)
      ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement[arm64][68](APMMeasurement+URL.o)
  _pb_decode, referenced from:
      _apmpb_decode_ad_campaign_info in GoogleAppMeasurement[arm64][79](apmpb_ad_campaign_info_decoder.o)
      _apmpb_decode_app_property in GoogleAppMeasurement[arm64][82](apmpb_app_property_decoder.o)
      _apmpb_decode_audience in GoogleAppMeasurement[arm64][85](apmpb_audience_decoder.o)
      _apmpb_decode_audience in GoogleAppMeasurement[arm64][85](apmpb_audience_decoder.o)
      _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][88](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][88](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_client_property in GoogleAppMeasurement[arm64][91](apmpb_client_property_decoder.o)
      _apmpb_decode_client_property in GoogleAppMeasurement[arm64][91](apmpb_client_property_decoder.o)
      ...
  _pb_decode_varint, referenced from:
      _nano_decode_repeated_uint64 in GoogleAppMeasurement[arm64][320](decoder_common.o)
  _pb_encode, referenced from:
      _nano_two_pass_encoding in GoogleAppMeasurement[arm64][321](encoder_common.o)
      _nano_two_pass_encoding in GoogleAppMeasurement[arm64][321](encoder_common.o)
  _pb_encode_string, referenced from:
      _nano_encode_string in GoogleAppMeasurement[arm64][321](encoder_common.o)
  _pb_encode_submessage, referenced from:
      _apmpb_encode_repeated_ad_campaign_info in GoogleAppMeasurement[arm64][80](apmpb_ad_campaign_info_encoder.o)
      _apmpb_encode_repeated_app_property in GoogleAppMeasurement[arm64][83](apmpb_app_property_encoder.o)
      _apmpb_encode_repeated_audience in GoogleAppMeasurement[arm64][86](apmpb_audience_encoder.o)
      _apmpb_encode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][89](apmpb_audience_leaf_filter_result_encoder.o)
      _apmpb_encode_repeated_client_property in GoogleAppMeasurement[arm64][92](apmpb_client_property_encoder.o)
      _apmpb_encode_repeated_consent_config in GoogleAppMeasurement[arm64][95](apmpb_consent_config_encoder.o)
      _apmpb_encode_repeated_consent_default_entry in GoogleAppMeasurement[arm64][98](apmpb_consent_default_entry_encoder.o)
      ...
  _pb_encode_tag_for_field, referenced from:
      _apmpb_encode_repeated_ad_campaign_info in GoogleAppMeasurement[arm64][80](apmpb_ad_campaign_info_encoder.o)
      _apmpb_encode_repeated_app_property in GoogleAppMeasurement[arm64][83](apmpb_app_property_encoder.o)
      _apmpb_encode_repeated_audience in GoogleAppMeasurement[arm64][86](apmpb_audience_encoder.o)
      _apmpb_encode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][89](apmpb_audience_leaf_filter_result_encoder.o)
      _apmpb_encode_repeated_client_property in GoogleAppMeasurement[arm64][92](apmpb_client_property_encoder.o)
      _apmpb_encode_repeated_consent_config in GoogleAppMeasurement[arm64][95](apmpb_consent_config_encoder.o)
      _apmpb_encode_repeated_consent_default_entry in GoogleAppMeasurement[arm64][98](apmpb_consent_default_entry_encoder.o)
      ...
  _pb_encode_varint, referenced from:
      _nano_encode_repeated_uint64 in GoogleAppMeasurement[arm64][321](encoder_common.o)
  _pb_istream_from_buffer, referenced from:
      _apmpb_decode_ad_campaign_info in GoogleAppMeasurement[arm64][79](apmpb_ad_campaign_info_decoder.o)
      _apmpb_decode_app_property in GoogleAppMeasurement[arm64][82](apmpb_app_property_decoder.o)
      _apmpb_decode_audience in GoogleAppMeasurement[arm64][85](apmpb_audience_decoder.o)
      _apmpb_decode_audience in GoogleAppMeasurement[arm64][85](apmpb_audience_decoder.o)
      _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][88](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][88](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_client_property in GoogleAppMeasurement[arm64][91](apmpb_client_property_decoder.o)
      _apmpb_decode_client_property in GoogleAppMeasurement[arm64][91](apmpb_client_property_decoder.o)
      ...
  _pb_ostream_from_buffer, referenced from:
      _nano_two_pass_encoding in GoogleAppMeasurement[arm64][321](encoder_common.o)
  _pb_read, referenced from:
      _apmpb_decode_repeated_ad_campaign_info in GoogleAppMeasurement[arm64][79](apmpb_ad_campaign_info_decoder.o)
      _apmpb_decode_repeated_ad_campaign_info in GoogleAppMeasurement[arm64][79](apmpb_ad_campaign_info_decoder.o)
      _apmpb_decode_repeated_app_property in GoogleAppMeasurement[arm64][82](apmpb_app_property_decoder.o)
      _apmpb_decode_repeated_app_property in GoogleAppMeasurement[arm64][82](apmpb_app_property_decoder.o)
      _apmpb_decode_repeated_audience in GoogleAppMeasurement[arm64][85](apmpb_audience_decoder.o)
      _apmpb_decode_repeated_audience in GoogleAppMeasurement[arm64][85](apmpb_audience_decoder.o)
      _apmpb_decode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][88](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][88](apmpb_audience_leaf_filter_result_decoder.o)
      ...
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by ncooke3** [comment]
Ok, this looks to be working. I won't merge since we need to replace all of the RCs with signed artifacts. I can do so with a zip from tonight's build.

File: Package.swift [hunk 1]
```diff
@@ -1359,6 +1359,7 @@ func abseilDependency() -> Package.Dependency {
       "https://github.com/google/abseil-cpp-binary.git",
       "1.2024011601.1" ..< "1.2024011700.0"
     )
+    return .package(url: packageInfo.url, branch: "nc/firestore-pre_rc1")
```
Comment (ncooke3) [inline Package.swift:1362]: Reminder to raise minimum supported version for Abseil. Probably should do this in grpc-binary as well.

**Pull request closed without merging**